x-pack/plugins/security_solution/common/endpoint/service/trusted_apps/validations.ts

```diff
@@ -34,6 +34,36 @@ export const getDuplicateFields = (entries: ConditionEntry[]) => {
     .map((entry) => entry[0]);
 };
 
+const WIN_EXEC_PATH = /\\(\w+\.\w+)$/i;
+const UNIX_EXEC_PATH = /\/\w+\.*\w*$/i;
+
+export const getExecutableName = ({
+  os,
+  value,
+}: {
+  os: OperatingSystem;
+  value: string;
+}): string => {
+  const execName =
+    os === OperatingSystem.WINDOWS ? value.match(WIN_EXEC_PATH) : value.match(UNIX_EXEC_PATH);
+  return execName ? execName[0].replaceAll(/\/|\\/gi, '') : '';
```

Comment:
There's the same problem here a filename in linux could have, for instance a backslash "\" and this regex will modify that name: e.g. of a valid unix filename: `/this/is/my/process/hola\\me`

Reply:
Slightly better regex 3922b0f29a36ac8dbc20221fbcd3f7a26d1bc5bc. I've created a ticket to update this later.
elastic/security-team/issues/2486